infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/PluginBrokerManager.java

```diff
@@ -69,7 +73,8 @@ public PluginBrokerManager(
       WorkspaceVolumesStrategy volumesStrategy,
       BrokerEnvironmentFactory<E> brokerEnvironmentFactory,
       UnrecoverablePodEventListenerFactory unrecoverablePodEventListenerFactory,
-      @Named("che.workspace.plugin_broker.wait_timeout_min") int pluginBrokerWaitingTimeout) {
+      @Named("che.workspace.plugin_broker.wait_timeout_min") int pluginBrokerWaitingTimeout,
+      @Nullable OptionalTracer tracer) {
```

Comment:
It can not be null - it is class. It WILL be instantiated by Guice on demand if nothing is bound. Apart from that it doesn't make sense to use this class AFAIK since you can just bind optional `Tracer`. Please, clean this up - I'm not sure I understand why this helper is needed at all.

Reply:
It seems this is part of the tracing framework I didn't really understand. Still don't think it is clean, but OK to leave it as is since it is already use in other places

Reply:
Yeah I don't understand it either, but it was the way it's done in other areas. AFAICT it's just `Optional<Tracer>` which I agree seems not useful.

Reply:
We had a discussion in platform team chat https://chat.openshift.io/developers/pl/tda3ybd8epfsmgewmgi1uhhb9c